iOS_SDK/OneSignalSDK/OneSignalUser/Source/OSUserRequests.swift

```diff
@@ -713,28 +835,42 @@ class OSRequestTransferSubscription: OneSignalRequest, OSUserRequest {
  Currently, only the Push Subscription will make this Update Request.
  */
 class OSRequestUpdateSubscription: OneSignalRequest, OSUserRequest {
+    let stringDescription: String
+    override var description: String {
+        return stringDescription
+    }
+
     let subscriptionModel: OSSubscriptionModel
 
     // Need the subscription_id
     func prepareForExecution() -> Bool {
-        if let subscriptionId = subscriptionModel.subscriptionId {
-            self.path = "subscriptions/\(subscriptionId)"
+        if let subscriptionId = subscriptionModel.subscriptionId, let appId = OneSignalConfigManager.getAppId() {
+            self.path = "apps/\(appId)/subscriptions/\(subscriptionId)"
             return true
         } else {
             self.path = "" // self.path is non-nil, so set to empty string
             return false
         }
     }
 
+    // TODO: just need the sub model and send it
+    // But the model may be outdated or not sync with the subscriptionObject
     init(subscriptionObject: [String: Any], subscriptionModel: OSSubscriptionModel) {
         self.subscriptionModel = subscriptionModel
+        self.stringDescription = "OSRequestUpdateSubscription with subscriptionObject: \(subscriptionObject)"
         super.init()
 
         // Rename "address" key as "token", if it exists
         var subscriptionParams = subscriptionObject
         subscriptionParams.removeValue(forKey: "address")
-        subscriptionParams["token"] = subscriptionObject["address"]
-
+        subscriptionParams.removeValue(forKey: "notificationTypes")
+        subscriptionParams["token"] = subscriptionModel.address ?? ""
+        subscriptionParams["notification_types"] = subscriptionModel.notificationTypes
+        subscriptionParams["enabled"] = subscriptionModel.enabled
+        // TODO: The above is not quite right. If we hydrate, we will over-write any pending updates
+//        subscriptionParams["token"] = subscriptionObject["address"]
```

Comment:
Remove commented out code if not needed.

Reply:
> Remove commented out code if not needed.

Reply:
Done, replaced with better comment to address the TODO mentioned.

Reply:
![image](https://user-images.githubusercontent.com/55927424/208774283-4fbead12-a6b9-43e4-82e6-48e460a15e81.jpg)